src/main/java/org/mifos/phee/kafkastreamer/importer/service/MaskingServiceImpl.java

```diff
@@ -0,0 +1,126 @@
+            log.info("Inside CHANNEL_GSMA_REQUEST condition");
+            String valueStringifiedJsonString = value.getString(KafkaVariables.VALUE);
+            JSONObject channelGsmaRequest = getJsonObjectFromStringifiedJson(valueStringifiedJsonString);
+
+            String debitIdentifier = channelGsmaRequest.getJSONArray(KafkaVariables.DEBIT_PARTY)
+                    .getJSONObject(0).getString(KafkaVariables.VALUE);
+            String creditIdentifier = channelGsmaRequest.getJSONArray(KafkaVariables.CREDIT_PARTY)
+                    .getJSONObject(0).getString(KafkaVariables.VALUE);
+
+            debitIdentifier = encryptData(debitIdentifier);
+            creditIdentifier = encryptData(creditIdentifier);
+
+            channelGsmaRequest.getJSONArray(KafkaVariables.DEBIT_PARTY)
+                    .getJSONObject(0).put(KafkaVariables.VALUE, debitIdentifier);
+            channelGsmaRequest.getJSONArray(KafkaVariables.CREDIT_PARTY)
+                    .getJSONObject(0).put(KafkaVariables.VALUE, creditIdentifier);
+
+            value.put(KafkaVariables.VALUE, channelGsmaRequest.toString());
+        } else {
+            log.info("Inside ELSE condition");
+            for (String field: fieldsToBeMasked) {
+                log.info("Field: {}", field);
+                if (name.equalsIgnoreCase(field)) {
+                    log.info("Field matched: {}", name);
+                    String fieldValue = value.getString(KafkaVariables.VALUE);
+                    fieldValue = encryptData(fieldValue);
+                    value.put(KafkaVariables.VALUE, fieldValue);
+                }
+            }
+        }
+
+        data.put(KafkaVariables.VALUE, value);
+        return data.toString();
+    }
+
+    // un-stringifies and returns the [JSONObject] instance from string
+    private JSONObject getJsonObjectFromStringifiedJson(String stringifiedJson) {
+        String s = unescapeJava(stringifiedJson);
+        s = s.substring(1, s.length()-1);
+        return new JSONObject(s);
+    }
+
+    // encrypts the data based on the length of the data passed
+    private String encryptData(String data) throws Exception {
+        String masked = "";
+        String unmasked = "";
+        if (data.isEmpty()) {
+            // nothing to mask
+            return data;
+        } else if (data.length() == 2) {
+             // leave last digit and mask first
+            unmasked = data.substring(data.length()-1);
+            masked = AesUtil.encrypt(data.substring(0, data.length()-1), encryptionKey);
+        } else if (data.length() == 1) {
+            // mask everything
+            masked =  AesUtil.encrypt(data, encryptionKey);
+        } else {
+            // leave last 2 digit and mask rest of it
+            unmasked = data.substring(data.length()-2);
+            masked = AesUtil.encrypt(data.substring(0, data.length()-2), encryptionKey);
+        }
+        return "{" + masked + "}" + unmasked;
```

Comment:
Use string builder for concatenation

Reply:
I used stringBuilder only but intellij auto code review suggested not to use stringBuilder for simpler and smaller concatenation.

Reply:
Thats a good one. I was thinking that this operation will repeat so many times. so it might be useful to do it with Stringnbuilder than String.